### PR TITLE
Remove is_final_block column in KeccakSpongeStark

### PR DESCRIPTION
**Pull request by Nashtare**
Following Daniel's suggestion in #950.

### CODE REVIEW

File: evm/src/keccak_sponge/keccak_sponge_stark.rs [hunk 1]
```diff
@@ -479,7 +473,13 @@ impl<F: RichField + Extendable<D>, const D: usize> Stark<F, D> for KeccakSpongeS
         );
         yield_constr.constraint(builder, constraint);
 
-        let is_final_block = local_values.is_final_block;
+        let mut is_final_block = builder.add_extension(
```
Comment (dlubarov) [inline evm/src/keccak_sponge/keccak_sponge_stark.rs:476]: I believe starting with an accumulator of zero should give the same performance; addition with zero should be skipped by `arithmetic_extension_special_cases`. There's also an `add_many_extension` method we could use for brevity.

File: evm/src/keccak_sponge/keccak_sponge_stark.rs [hunk 2]
```diff
@@ -382,12 +379,8 @@ impl<F: RichField + Extendable<D>, const D: usize> Stark<F, D> for KeccakSpongeS
         // Ensure that full-input block and final block flags are not set to 1 at the same time.
         yield_constr.constraint(is_final_block * is_full_input_block);
```
Comment (dlubarov) [inline evm/src/keccak_sponge/keccak_sponge_stark.rs:380]: Looks like this is now the same as the next constraint below, we can remove one or the other :)

**Review by dlubarov — APPROVED** [review]
LGTM otherwise, nice simplification

File: evm/src/keccak_sponge/keccak_sponge_stark.rs (hunk 1 above)
Comment (Nashtare) [reply evm/src/keccak_sponge/keccak_sponge_stark.rs]: I opted for `add_many_extension`, thanks for the tip.

File: evm/src/keccak_sponge/keccak_sponge_stark.rs (hunk 2 above)
Comment (Nashtare) [reply evm/src/keccak_sponge/keccak_sponge_stark.rs:380]: Updated!

**Pull request merged**